src/main/java/il/ac/bgu/cs/bp/bpjs/model/BProgram.java

```diff
@@ -504,7 +507,15 @@ public void setLogLevel( BpLog.LogLevel aLevel ) {
             preSetLogLevel = aLevel;
         }
     }
-    
+
+    /**
+     * Set the log implementation - logging called below the set level are logged using @BpLog implementation.
+     * @param logger The new logger implementation to use.
+     */
+    public void setLogger(BpLog logger){
+        preSetLogger = logger;
```

Comment:
If I understand, once the BProgram is started, this set would have no effect. So in this case we also need to update the logger in `jsProxy`. (unless I'm missing something here).

Reply:
we cannot change the log after initializing the b-program, I added an exception.